actix/src/io.rs

```diff
@@ -526,8 +526,11 @@ where
                         break;
                     }
                 }
-                Poll::Ready(Err(_err)) => {
-                    break;
+                Poll::Ready(Err(e)) => {
+                    if act.error(e, ctxt) == Running::Stop {
```

Comment:
Ideally it should not be allowed to *not* stop at this point (and below in the error cases). Polling the `Sink` again after errors is generally a bad idea.